app/services/business_entity_service.rb

```diff
@@ -0,0 +1,56 @@
+class BusinessEntityService
```

Comment:
Good point by @novotnyjakub. I was really worried that I wasn't getting the existence of both:

* @business_entity 
* business_entity

Renaming them into something so that they differ would be much easier to read and follow.